### PR TITLE
Remove remaining uses of `originalMessage.participants` as part of personalDetails migration

### PR DESCRIPTION
**Pull request by Beamanator**
### Details

This PR removes/replaces all use of `reportActions_*.originalMessage.participants`. This is part of the email -> accountID migration. 
Note: This is NOT dealing with `report_*.participants`!! Only `reportActions_*.originalMessage.participants` 👍 

### Fixed Issues
$ Related to https://github.com/Expensify/App/issues/19007

### Tests
Luckily we don't check this often, so the only places we need to test are:
1. Creating a money request with users you already know and have chats with 
2. Splitting bills with users you already know and have chats with 

Note: splitting bills with users you don't already have chats with seems to be broken, but it also seems to be broken on main. 

- [ ] Verify that no errors appear in the JS console

### Offline tests

Same as online tests. 

### QA Steps

Same as tests. 

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-07-06 at 2 42 47 PM
Screenshot 2023-07-06 at 2 41 32 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-07-06 at 4 54 09 PM

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-07-06 at 2 58 16 PM

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-07-06 at 4 57 33 PM

</details>

<details>
<summary>Android</summary>

Screenshot 2023-07-06 at 4 52 12 PM

</details>

### CODE REVIEW

File: src/pages/iou/SplitBillDetailsPage.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ function getReportID(route) {
 
 function SplitBillDetailsPage(props) {
     const reportAction = props.reportActions[`${props.route.params.reportActionID.toString()}`];
-    const participantAccountIDs = reportAction.originalMessage.participantAccountIDs || PersonalDetailsUtils.getAccountIDsByLogins(reportAction.originalMessage.participants);
+    const participantAccountIDs = reportAction.originalMessage.participantAccountIDs;
```
Comment (Beamanator) [inline src/pages/iou/SplitBillDetailsPage.js:68]: @puneetlath I thinkkkk we're only going to be able to do this once we finish the DB migration - https://github.com/Expensify/Expensify/issues/292074

Comment (puneetlath) [reply src/pages/iou/SplitBillDetailsPage.js:67]: I think we should go ahead and let old IOUs break. We should clean them up, but we don't need to block this on it.

**Review by 0xmiroslav — COMMENTED** [review]
Tests well so far. Will checklist tomorrow morning

**Comment by puneetlath** [comment]
working on it now!

File: src/libs/ReportUtils.js [hunk 2]
```diff
@@ -1446,7 +1446,6 @@ function buildOptimisticIOUReportAction(type, amount, currency, comment, partici
     // IOUs of type split only exist in group DMs and those don't have an iouReport so we need to delete the IOUReportID key
     if (type === CONST.IOU.REPORT_ACTION_TYPE.SPLIT) {
         delete originalMessage.IOUReportID;
-        originalMessage.participants = [currentUserEmail, ..._.pluck(participants, 'login')];
```
Comment (stitesExpensify) [inline src/libs/ReportUtils.js:1449]: Just to double check, we're intentionally not setting this anymore?

**Review by stitesExpensify — APPROVED** [review]
Code LGTM, just one question

File: src/libs/ReportUtils.js (hunk 2 above)
Comment (puneetlath) [reply src/libs/ReportUtils.js:1449]: That's correct. We set `participantAccountIDs` instead on the line below.

File: src/pages/iou/SplitBillDetailsPage.js (hunk 1 above)
Comment (Beamanator) [reply src/pages/iou/SplitBillDetailsPage.js:67]: Alright i think that's fair - then let's do this!! 👍  And the DB migration is ready to go at a moment's notice anyway so we can get this to production & finish the migration quick 👍

**Comment by Beamanator** [comment]
@0xmiroslav thanks so much for you testing, I'm going to look into some of these issues on main shortly - but for this PR I think we can move forward, there's basically only 1 place we're truly removing changing things (moving our use of `originalMessage.participants` to `originalMessage.participantAccountIDs`) - in the split bill details page. So if that still looks OK I would say let's move forward

**Pull request merged**